### DISCUSSION TITLE
Installation Issue

### DISCUSSION THREAD

**Discussion by JW-vfx** [danilodelucio/NukeDiffusion#1] [General]
Hi. Getting the following error when running it from Nuke 15.0v4.
- It was not possible to generate your image!
- Error: No package metadata was found for The 'filelock' distribution was not found and is required by this
application.
Try: pip install transformers -U or pip install -e '.' if you're working with git main.

'Filelock' is installed 
Requirement already satisfied: filelock in ........\appdata\local\programs\python\python312\lib\site-packages (3.13.1)

Any idea why this errors?
I have also tried several different models but still does not work.

**Comment by danilodelucio** [comment]
Hello... few questions:

1. Do the Checkpoints are matching to the SD model? Make sure the version you are using of the Checkpoint is SD or SDXL, otherwise will not work;
2. What resolution and workflow are you using?;
3. Did you try using the `SD Default Model` option as well?

By the way, the `Filelock` seems to be installed in your system, which is the wrong path.
Please try to install this `transformers` module in the Python interpreter from NukeDiffusion, go to `.\NukeDiffusion\python\python3.11.6` and open your terminal from there (right click on empty space -> Open in Terminal or click on URL of your Explorer and type `cmd`), then run the following code:
```
python.exe -m pip install transformers -U
```

Let me know if this works!

**Reply by JW-vfx** [reply]
Thanks for the fast response.
The checkpoints do match SD or SDXL, or auto switches as well which is nice.
Resolution is default 512x512.
Getting the same error on the SD Default Model as well.
When installing the transforms module on the .nuke folder as per your instruction, it says 'Requirement already satisfied'. But still points to the 'appdata' folder though.

**Reply by danilodelucio** [reply]
No problem!

Did you try to install the `transformers` module like this?

![image](https://github.com/danilodelucio/NukeDiffusion/assets/47226196/dbadc5d3-1ab5-4f70-8a1e-a98cdedcec62)

Unfortunately I don't know this issue, seems to be a missing module from Hugging Face (the `filelock` module).
If possible send me some screenshots to try to assist you.

**Reply by JW-vfx** [reply]
Hi. Thanks again for the reply. I have done as you suggested, still getting a 'Requirement already satisfied'
![image_2024-03-12_165947566](https://github.com/danilodelucio/NukeDiffusion/assets/72604191/56d17210-85f2-49e0-b6a6-490521333f21)
Even with the corrected path. Had to adjust the install code to '.\python.exe -m pip install transformers -U', but other than that everything seems to be in order. 
Weird that it isn't working.

**Reply by danilodelucio** [reply]
Yeah this looks weird!

I see that the NukeDiffusion folder is named as **NukeDiffusion-main**. In the `init.py` from the `.nuke` folder, are you pointing the same name as `nuke.pluginAddPath('./Python/NukeDiffusion-main')`?

I've helped a few people and all of them had the same issue, and it's my bad because I didn't remember that when you download the NukeDiffusion the folder comes with **-main** at the end.

In summary:

- If your folder is named as **NukeDiffusion-main**, your init file should be `nuke.pluginAddPath('./NukeDiffusion-main')`;
- If your folder is named as **NukeDiffusion**, your init file should be `nuke.pluginAddPath('./NukeDiffusion')`;

Probably this is not your case because you said earlier that you were trying different Checkpoints (but just making sure).

Let's try another approach...

Since your first error is regarding the `filelock` module, let's try to install it directly. Before that, check if it exists by running the following code:

```
python.exe -m pip list | findstr filelock
```
![image](https://github.com/danilodelucio/NukeDiffusion/assets/47226196/216a4597-8844-4df7-a101-692f410b4ab0)

If doesn't exist, then try:
```
python.exe -m pip install filelock
```
![image](https://github.com/danilodelucio/NukeDiffusion/assets/47226196/563b7c61-1f98-4ccf-aa24-3690784cc1ac)

I believe you accidentally closed Terminal before the libraries were installed (just guessing).

Well, let me know if this works! 🤞🍀
